scipy/interpolate/_ndbspline.py

```diff
@@ -251,7 +250,11 @@ def design_matrix(cls, xvals, t, k, extrapolate=True):
                 f"Data and knots are inconsistent: len(t) = {len(t)} for "
                 f" {ndim = }."
             )
-        # XXX: other consistency checks
+        try:
+            len(k)
+        except TypeError:
+            # make k a tuple
+            k = (k,)*ndim
```

Comment:
Just to make sure I understand--`k` is documented as/supposed to be an int, and you're generally expecting the `except` path to be used here for an internal conversion only?

Reply:
Exactly! Internal machinery wants an array of orders in each dimension (wanting a cubics in the direction x times a quadratic in the direction y is supported if somewhat exotic). For a casual user it is probably reasonable to be able supply a single scalar to mean "cubics in all directions". So a better description of the argument is "int or array_like, shape(ndim,)".

The EAFP pattern matches the one used in the main constructor.